packages/application-shell/src/components/application-shell/application-shell.tsx

```diff
@@ -141,336 +141,345 @@ export const RestrictedApplication = <
     Props<AdditionalEnvironmentProperties>,
     'onRegisterErrorListeners'
   >
-) => (
-  <FetchUser>
-    {({ isLoading: isLoadingUser, user, error }) => {
-      if (error) {
-        // In case there is an unauthorized error, we redirect to the login page
-        if (error.graphQLErrors && Array.isArray(error.graphQLErrors)) {
-          const hasUnauthorizedError = getHasUnauthorizedError(
-            error.graphQLErrors
-          );
-          const hasUserBeenDeletedError = getHasUserBeenDeletedError(
-            error.graphQLErrors
-          );
-
-          if (hasUnauthorizedError || hasUserBeenDeletedError) {
-            let logoutReason:
-              | typeof LOGOUT_REASONS[keyof typeof LOGOUT_REASONS]
-              | undefined;
-            if (hasUnauthorizedError)
-              logoutReason = LOGOUT_REASONS.UNAUTHORIZED;
-            else if (hasUserBeenDeletedError)
-              logoutReason = LOGOUT_REASONS.DELETED;
-            return (
-              <Redirector to="logout" queryParams={{ reason: logoutReason }} />
+) => {
+  const location = useLocation();
```

Comment:
Here we read the location

Reply:
FYI: So this is a bit of a tech debt. I commented above what the trade-off is. There is no "easy" way not to achieve what we want to without having to subscribe to route changes.
Maybe if we rewrite the component to use hooks and split it up a bit, we might be able to "reduce" the area of influence regarding re-rendering when a route changes.
We should also look out a bit for performance, in case we notice that the app got "slower" because of the many re-rendering.